File: utility/aws/golang/utility/var.go
// © Copyright IBM Corporation 2020. All rights reserved.
// SPDX-License-Identifier: Apache2.0
// 
package utility

type CredentialInfo struct {
	Environment string
	Domain      string
	Service     string
	Variable    string
}

type ParameterContent struct {
	Value       string
	Description string
}

type SecretContent struct {
	Entry       []SecretEntry
	Description string
	FilePath    string
	RawJson     []byte
}

type SecretEntry struct {
	Key   string
	Value string
}

File: utility/aws/javascript/src/utility/var.ts
// © Copyright IBM Corporation 2020. All rights reserved.
// SPDX-License-Identifier: Apache2.0
// 
export interface CredentialInfo {
	environment: string;
	domain: string;
	service: string;
	variable: string;
}

export interface ParameterContent {
	value: string;
	description: string;
}

export interface SecretContent {
	key?: string;
	value?: string;
	description: string;
	filePath?: string;  //absolute path
}

export const AWS_SECRET = "AWS"

export const LOCAL_SECRET = "LOCAL"

export const VAULT_SECRET = "VAULT"